src/proxy/inbound.rs

```diff
@@ -107,41 +108,61 @@ impl Inbound {
             }
         }
     }
+
+    /// handle_inbound serves an inbound connection with a target address `addr`.
+    pub(super) async fn handle_inbound(
+        request_type: InboundConnect,
+        addr: SocketAddr,
+    ) -> Result<(), std::io::Error> {
+        let stream = tokio::net::TcpStream::connect(addr).await;
+        match stream {
+            Err(err) => {
+                warn!("connect to {} failed: {}", addr, err);
+                Err(err)
+            }
+            Ok(stream) => {
+                let mut stream = stream;
+                tokio::task::spawn(async move {
+                    match request_type {
+                        InboundConnect::DirectPath(mut incoming) => {
```

Comment:
nit: can we move the `match request_type` into an impl on the `InboundConnect` enum to avoid the nesting?